src/array/mod.rs

```diff
@@ -244,7 +254,8 @@ macro_rules! impl_from {
 
                 fn try_from(array: ArrayImpl) -> Result<Self, Self::Error> {
                     match array {
-                        ArrayImpl::$Abc(array) => Ok(array),
+                        // ArrayImpl::$Abc(array) => Ok(array),
+                        ArrayImpl::$Abc(array) => loop {},
```

Comment:
I think this function should be implemented on `Arc<Array>` then. `ArrayImpl::Bool` can now only be turned into an `Arc<BoolArray>`.

Reply:
Done, thanks for suggesting this.